tests/ui/needless_character_iteration.rs

```diff
@@ -0,0 +1,26 @@
+#![warn(clippy::needless_character_iteration)]
+#![allow(clippy::unnecessary_operation)]
+
+fn main() {
+    "foo".chars().all(|c| c.is_ascii());
+    //~^ ERROR: checking if a string is ascii using iterators
+    "foo".chars().all(|c| !c.is_ascii());
+    //~^ ERROR: checking if a string is ascii using iterators
+
+    let s = String::new();
+    s.chars().all(|c| c.is_ascii());
+    //~^ ERROR: checking if a string is ascii using iterators
+    "foo".to_string().chars().all(|c| !c.is_ascii());
+    //~^ ERROR: checking if a string is ascii using iterators
+
+    "foo".chars().all(|c| {
+        //~^ ERROR: checking if a string is ascii using iterators
+        let x = c;
+        x.is_ascii()
+    });
```

Comment:
Could you also add a test where the closure contains some additional statements, like this:
```
    "foo".chars().all(|c| {
        let x = c;
        magic(x);
        x.is_ascii()
    });
```

Reply:
Sure!

Reply:
That was a very good call, the lint was emitted in this case but shouldn't have! Fixed it. :)